Check if branch exists on remote

For me the action is currently failing if the selected branch does not exist on the remote. 


```
fatal: couldn't find remote ref gh-pages
fatal: No pathspec was given. Which files should I remove?
Error: Process completed with exit code 123.
```

Maybe it there should be a check if the branch exist? 

At the moment the test is also not failing if there is no branch. For this I created a separate test [Test - branch not existing ](https://github.com/t-huyeng/push-dir/actions/runs/3211278652/jobs/5249335349) even during the runtime of the action the same errors occur.

--- a/.github/workflows/comment-bot.yml
+++ b/.github/workflows/comment-bot.yml
@@ -9,7 +9,7 @@ jobs:
     if: startsWith(github.event.comment.body, '/tag ')
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: React Seen
       uses: actions/github-script@v2
       with:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -8,7 +8,10 @@ jobs:
   test:
     if: github.event_name != 'pull_request' || !contains('OWNER,MEMBER,COLLABORATOR', github.event.pull_request.author_association)
     runs-on: ubuntu-latest
-    name: Test
+    name: Test-${{ matrix.remote-branch }}
+    strategy:
+      matrix:
+        remote-branch: [0, 1]
     steps:
     - uses: actions/checkout@v3
     - name: setup
@@ -23,13 +26,15 @@ jobs:
           git add .
           git commit -m "initial commit"
 
-          git checkout -b gh-pages
-          rm *
-          touch old
-          touch updated
-          git add .
-          git commit --amend --no-edit
-          git checkout -
+          if test ${{ matrix.remote-branch }} -eq 1; then
+            git checkout -b gh-pages
+            rm *
+            touch old
+            touch updated
+            git add .
+            git commit --amend --no-edit
+            git checkout -
+          fi
         popd
 
         mkdir test-repo

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     name: Test
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: setup
       run: |
         git config --global init.defaultBranch main

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ Cleanly push directory contents to a branch. Particularly useful for `gh-pages` 
 
 ```yaml
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - run: build_site_command --output-dir static_site
       - uses: casperdcl/push-dir@v1
         with:

--- a/action.yml
+++ b/action.yml
@@ -69,8 +69,8 @@ runs:
       if [[ "$INPUT_HISTORY" == true ]]; then
         git -C "$DOT_GIT_DIR" fetch --depth=1 origin "$INPUT_BRANCH":"$INPUT_BRANCH" \
         && git fetch --depth=1 local "$INPUT_BRANCH":"$INPUT_BRANCH" \
-        && git checkout "$INPUT_BRANCH" || :
-        git ls-files -z | xargs -0 git rm
+        && git checkout "$INPUT_BRANCH" \
+        && git ls-files -z | xargs -0 git rm || :
       fi
       git branch -M "$INPUT_BRANCH"
 

